### PR TITLE
32 http retries and error logging/handling improvements

### LINKED ISSUE #32: Properly resolve errors during db write operations
## 💡 Description

#29 implemented log-and-continue for db write failures as an ASAP fix for #28 

Once it's confirmed that allows successful cloud execution against gamma, this must be reworked such that:

- retry-based resilience behaviour is implemented @tloubrieu-jpl @jordanpadams I recall there being a ticket relating to this, but I can't find it
- fatal error in db write (unsuccessful after retries exhausted) logs and fails fatally

### PR DESCRIPTION
**Pull request by alexdunnjpl**
## 🗒️ Summary
 - HTTP calls are now subject to retry with backoff (2/4/8sec each call)
 - db write errors are now logged as `WARNING` if expected, or `ERROR` if unexpected.  `ERROR` should be flagged for action in CloudWatch Alarms @sjoshi-jpl 
 - elapsed time is now displayed in logs at end of execution, for ease of use

## ⚙️ Test Data and/or Report
Unit tests pass, manually tested against prod

## ♻️ Related Issues
fixes #32

### CODE REVIEW

File: src/pds/registrysweepers/utils/__init__.py [hunk 1]
```diff
@@ -13,8 +14,11 @@
 from typing import Mapping
 from typing import Optional
 from typing import Union
+from urllib.error import HTTPError
 
 import requests
+from retry import retry
```
[the remark this answers is not in the document]
Comment (alexdunnjpl) [reply src/pds/registrysweepers/utils/__init__.py:20]: shut

File: src/pds/registrysweepers/utils/__init__.py [hunk 2]
```diff
@@ -13,8 +14,11 @@
...
+from retry.api import retry_call
```
[the remark this answers is not in the document]
Comment (alexdunnjpl) [reply src/pds/registrysweepers/utils/__init__.py:21]: it

File: docker/sweepers_driver.py [hunk 3]
```diff
@@ -58,10 +58,11 @@
 import json
 import logging
 import os
+from datetime import datetime
 from typing import Callable, Iterable
 
 from pds.registrysweepers import provenance, ancestry
-from pds.registrysweepers.utils import configure_logging
+from pds.registrysweepers.utils import configure_logging, get_human_readable_elapsed_since
```
[the remark this answers is not in the document]
Comment (alexdunnjpl) [reply docker/sweepers_driver.py:65]: you're

File: src/pds/registrysweepers/utils/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexdunnjpl) [reply src/pds/registrysweepers/utils/__init__.py:20]: retiring

File: src/pds/registrysweepers/utils/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexdunnjpl) [reply src/pds/registrysweepers/utils/__init__.py:21]: anyway

**Review by jordanpadams — APPROVED** [review]

File: src/pds/registrysweepers/utils/__init__.py [hunk 4]
```diff
@@ -250,3 +278,11 @@ def coerce_list_type(db_value: Any) -> List[Any]:
             db_value,
         ]
     )
+
+
+def get_human_readable_elapsed_since(begin: datetime) -> str:
```
Comment (nutjob4life) [inline src/pds/registrysweepers/utils/__init__.py:283]: `humanize` provides a nice library for this, but this works too

**Review by nutjob4life — APPROVED** [review]

**Comment by jordanpadams** [comment]
🚀

File: src/pds/registrysweepers/utils/__init__.py (hunk 4 above)
Comment (alexdunnjpl) [reply src/pds/registrysweepers/utils/__init__.py:283]: Seemed like a weak justification for a new dependency 😅

**Comment by alexdunnjpl** [comment]
Cheers lads

**Pull request merged**